### DISCUSSION TITLE
read property of metadata, when using OnAfterConversionToEntity event

### DISCUSSION THREAD

**Discussion by mic2zar** [ravendb/ravendb#12991] [Q&A] [answered]
How could I read document metadata property, when using OnAfterConversionToEntity event. I am looking for something along this line and could not find any close by example in the documentation:

            documentStore.OnAfterConversionToEntity += (sender, e) =>
            {
                var metadata = e.Document["@metadata"];
                var lastModifiedAt = metadata["@last-modified"]; // this does not work
            };

**Comment by ppekrol** [comment, marked as answer]
Have you tried doing

```csharp
                    if (args.Document.TryGet("@metadata", out BlittableJsonReaderObject metadata))
                    {
                        if (metadata.TryGet("@last-modified", out DateTime lastModified))
                        {
                            // logic here
                        }
                    }
```

?
